staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go

```diff
@@ -120,7 +120,7 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 
 	if intoUnknown, ok := into.(*runtime.Unknown); ok && intoUnknown != nil {
 		*intoUnknown = unk
-		if ok, _, _ := s.RecognizesData(bytes.NewBuffer(unk.Raw)); ok {
+		if ok, _, _ := s.RecognizesData(unk.Raw); ok {
```

Comment:
cc @enj

Reply:
@smarterclayton pretty sure the existing code is just wrong, see #98439.

Reply:
Either way doesn't have anything to do with this change and is orthogonal (I will review that separately).